Fabric.Authorization.AccessControl/src/app/modules/access-control/custom-group-add/custom-group-add.component.spec.ts

```diff
@@ -9,11 +15,17 @@ describe('CustomGroupAddComponent', () => {
   beforeEach(
     async(() => {
       TestBed.configureTestingModule({
-        declarations: [CustomGroupAddComponent]
+        declarations: [CustomGroupAddComponent],
+        imports: [ServicesMockModule, FormsModule],
+        providers: [FabricExternalIdpSearchService, HttpClient, HttpHandler, AccessControlConfigService]
       }).compileComponents();
     })
   );
 
+  beforeEach(inject([FabricAuthRoleService], (roleService: FabricAuthRoleServiceMock) => {
+    roleService.getRolesBySecurableItemAndGrain.and.returnValue(Observable.of(mockRoles));
```

Comment:
The mocks just have an empty spy for each method as not to opinionate anything. Each service call will need access the spy and return their own results. I did export a default mocked result so it is easier to work with

Reply:
This is a standard we are doing in atlas currently